src/datasets/download/streaming_download_manager.py

```diff
@@ -442,11 +405,38 @@ def _get_extraction_protocol(urlpath: str, token: Optional[Union[str, bool]] = N
             raise
 
 
-def _prepare_http_url_kwargs(url: str, token: Optional[Union[str, bool]] = None) -> Tuple[str, dict]:
+def _validate_servers(urlpath: str):
+    server = urlpath.split("://")[0]
+    return server in SUPPORTED_REMOTE_SERVER_TYPE
+
+
+def _prepare_server_config(
+    path: str, download_config: Optional[DownloadConfig] = None, implemented: bool = True
+) -> Tuple[str, dict]:
+    main_hop, *rest_hops = str(path).split("::")
+    if not rest_hops and _validate_servers(main_hop):
+        if not implemented:
+            raise NotImplementedError("Currently not extended to support URLs in streaming mode")
+        main_hop, http_kwargs = _prepare_http_url_kwargs(main_hop, download_config=download_config)
```

Comment:
_prepare_http_url_kwargs should only be called on https URLs no ?

And for any other fsspec URI you can return the path unchanged and the storage_options from download_config

Reply:
yes, _prepare_http_url_kwargs should only be called for URLs where we need to set the kwargs.
that really makes the code clearer and more elegant.